multisig-wallet/src/main.sw

```diff
@@ -1,172 +1,195 @@
-    /// - When the signer is not an owner
     /// - When the recovered addresses are not in ascending order (0x1 < 0x2 < 0x3...)
     /// - When the total approval count is less than the required threshold for execution
-    fn execute_transaction(to: ContractId, value: u64, data: b256, signature1: B512, signature2: B512) -> bool {
-        require(storage.nonce != 0, Error::NotInitialized);
+    fn execute_transaction(to: Identity, value: u64, data: b256, signatures: [B512;
+    2]) {
+        require(storage.nonce != 0, InitError::NotInitialized);
 
-        let tx_hash = _get_transaction_hash(to, value, data, storage.nonce, contract_id());
+        let transaction_hash = create_hash(to, value, data, storage.nonce, contract_id());
+        let approval_count = count_approvals(transaction_hash, signatures);
 
-        // TODO: change to Vec<B512> once implemented and then iterate instead of hardcoding length
-        let signer1: b256 = match ec_recover_address(signature1, tx_hash) {
-            Result::Ok(address) => address.value, _ => revert(42), 
-        };
+        require(storage.threshold <= approval_count, ExecutionError::InsufficientApprovals);
 
-        let signer2: b256 = match ec_recover_address(signature2, tx_hash) {
-            Result::Ok(address) => address.value, _ => revert(42), 
-        };
-
-        require(~b256::min() < signer1 && signer1 < signer2, Error::IncorrectSignerOrdering);
-
-        let signer1_weight = get::<u64>(signer1);
-        let signer2_weight = get::<u64>(signer2);
-
-        require(signer1_weight != 0 && signer2_weight != 0, Error::NotAnOwner);
-
-        // Hardcoded value, that passes the checks above, until the loop below is unblocked
-        let approval_count = signer1_weight + signer2_weight;
-
-        // The signers must have increasing values in order to check for duplicates or a zero-value
-        // let mut previous_signer: b256 = ~b256::min();
+        storage.nonce = storage.nonce + 1;
 
-        // let mut approval_count = 0;
-        // let mut index = 0;
-        // while index < 2 {
-        //     let signer: b256 = match ec_recover_address(signatures[index], tx_hash) {
-        //         Result::Ok(address) => address.value, _ => revert(42),
-        //     };
+        // TODO: Execute https://github.com/FuelLabs/sway-applications/issues/22
 
-        //     require(previous_signer < signer, Error::IncorrectSignerOrdering);
+        log(ExecutedEvent {
+            to, value, data, nonce: storage.nonce - 1
+        });
+    }
 
-        //     let signer_weight = get::<u64>(signer);
-        //     require(signer_weight != 0, Error::NotAnOwner);
+    /// Transfers assets to outputs & contracts if the signatures meet the threshold requirement
+    ///
+    /// # Panics
+    ///
+    /// - When the constructor has not been called to initialize the contract
+    /// - When the balance of the asset being sent is less than the balance in the contract
```

Comment:
Shouldn't this be "When the balance of the asset in the contract is less than the amount being transferred." ?

Reply:
How about now?